test/asn1_internal_test.c

```diff
@@ -85,8 +85,10 @@ static int test_standard_methods(void)
          *
          * Anything else is an error and may lead to a corrupt ASN1 method table
          */
-        if (!TEST_int_eq(((*tmp)->pem_str == NULL),
-                         (((*tmp)->pkey_flags & ASN1_PKEY_ALIAS) != 0))) {
+        if (!TEST_true((*tmp)->pem_str == NULL &&
+                       ((*tmp)->pkey_flags & ASN1_PKEY_ALIAS) != 0)
+            && !TEST_true((*tmp)->pem_str != NULL &&
+                          ((*tmp)->pkey_flags & ASN1_PKEY_ALIAS) == 0)) {
```

Comment:
Why not use the same condition as above, i.e.   failure is !(good case) ?

```
if (!TEST_true((pem_str == NULL && (flags & ASN1_PKEY_ALIAS) != 0)
               ||(pem_str != NULL && (flags & ASN1_PKEY_ALIAS) == 0))
```

Reply:
Because I want it to be clear from the diagnostic which if the two alternatives failed.

Reply:
Ok, that's an argument.  Reconfirmed.

Reply:
OTOH, it's like an exlusive-or: if the second condition holds you will always get error diagnostics for the failed first condition, is this really intended?

Reply:
> Because I want it to be clear from the diagnostic which of the two alternatives failed.

Note: the `if(...)` test will fail if and only if _both_ `TEST_true(...)` tests fail. So the output does not give any additional information, only additional noise in the good case where the second `TEST_true(...)`  succeeds. See #6888.